src/frontend/src/lib/utils/tokens.utils.ts

```diff
@@ -40,3 +45,44 @@ export const sortTokens = ({
 			a.name.localeCompare(b.name) ||
 			a.network.name.localeCompare(b.network.name)
 	);
+
+export const pinTokensWithBalanceAtTop = ({
+	$tokens,
+	$exchanges,
+	$balancesStore
+}: {
+	$tokens: Token[];
+	$exchanges: ExchangesData;
+	$balancesStore: CertifiedStoreData<BalancesData>;
+}): Token[] => {
+	const tokensWithBalanceToPin: TokenToPin[] = $tokens
+		.map((token) => ({
+			...token,
+			usdValue: nonNullish($exchanges?.[token.id])
```

Comment:
- In which type is `usdValue` defined?
- If the `usdValue` is now used for sorting purpose as well, should we integrate in the global store and global type or should we keep it as an optional value calculated on the fly? For example for the same token we gonna access the exchange store to present the value in dollar to sort the token and then for render purpose - i.e. twice at least.

Reply:
- it is not defined for any type, that is why i put it in a concatenation mapping/filter/sorting
- I wanted to talk with you about that: I agree, since we are already calculating it twice, better to put it in a single place. My question is: in a PR before this one or in a PR after this one?

Reply:
on it

Reply:
check PR #1785 please

Reply:
adapted code to use the new prop